### PR TITLE
General Typescript linting and typing standardization

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR focuses on general npm typing and linting standardization which is necessary for general code quality, testing, and subsequent CI/CD actions.

- 4b5ebc1 Add unit test and codeql workflows
- 73ec3f8 Add some more stub top-level readme content
- 0aebf73 Mass linting, typing and unit test fixes
- de41c46 Standardize and sanitize eslint code style across codebase
- bf008d2 Update vscode plugin recommendations

[SHOWCASE-2886](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-2886)

## Types of changes

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)

Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/components/object/ObjectTable.vue [hunk 1]
```diff
@@ -30,26 +29,17 @@ const showInfo = async (id: string) => {
...
-    :value="objectList"
-    dataKey="id"
-    class="p-datatable-sm"
-    stripedRows
-    responsiveLayout="scroll"
-    :paginator="true"
-    :rows="10"
+  <DataTable :loading="loading" :value="objectList" dataKey="id" class="p-datatable-sm" stripedRows
+    responsiveLayout="scroll" :paginator="true" :rows="10"
     paginatorTemplate="RowsPerPageDropdown CurrentPageReport PrevPageLink NextPageLink "
```
Comment (TimCsaky) [inline frontend/src/components/object/ObjectTable.vue:34]: extra space

File: frontend/src/components/bucket/BucketsTable.vue [hunk 2]
```diff
@@ -30,19 +30,10 @@ const showPermissions = async (bucketId: string, bucketName: string) => {
...
-    :value="buckets"
-    dataKey="bucketId"
-    class="p-datatable-sm"
-    stripedRows
-    responsiveLayout="scroll"
-    :paginator="true"
-    :rows="10"
+  <DataTable :loading="loading" :value="buckets" dataKey="bucketId" class="p-datatable-sm" stripedRows
+    responsiveLayout="scroll" :paginator="true" :rows="10"
     paginatorTemplate="RowsPerPageDropdown CurrentPageReport PrevPageLink NextPageLink "
```
[the remark this answers is not in the document]
Comment (jujaga) [reply frontend/src/components/bucket/BucketsTable.vue:35]: Unfortunately the autoformatter doesn't seem to want to do multiple lines by itself. Not sure how to coerce it to do that. Manually fixing for now.

File: frontend/src/components/object/ObjectList.vue [hunk 3]
```diff
@@ -82,4 +80,6 @@ onMounted(() => {
   </div>
 </template>
 
-<style scoped></style>
+<style scoped>
+
```
[the remark this answers is not in the document]
Comment (jujaga) [reply frontend/src/components/object/ObjectList.vue:84]: Autoformatter behavior

File: frontend/src/components/object/ObjectTable.vue (hunk 1 above)
Comment (jujaga) [reply frontend/src/components/object/ObjectTable.vue:34]: Autoformatter behavior. Manually overriding.

File: frontend/src/store/userStore.ts [hunk 4]
```diff
@@ -6,33 +6,38 @@ export const useUserStore = defineStore('user', () => {
   const idps = ref([] as Object[]);
   const loading = ref(false);
 
+  // TODO: Implement stub function
```
[the remark this answers is not in the document]
Comment (jujaga) [reply frontend/src/store/userStore.ts:9]: No idea. I don't know where this is being used and in what context. I only focused on getting all the linting/typing issues resolved so this needs to be looked at later by someone that knows the code organization better.

File: frontend/src/store/userStore.ts [hunk 5]
```diff
@@ -6,33 +6,38 @@ export const useUserStore = defineStore('user', () => {
...
+      console.error(`bad error: ${error}`); // eslint-disable-line no-console
       // So that a caller can action it
       throw error;
     } finally {
       loading.value = false;
     }
   }
 
-  return { idps, loading, listIdps, testBad };
+  return { addBucket, idps, loading, listIdps, testBad };
```
[the remark this answers is not in the document]
Comment (jujaga) [reply frontend/src/store/userStore.ts:42]: No idea. I don't know where this is being used and in what context. I only focused on getting all the linting/typing issues resolved so this needs to be looked at later by someone that knows the code organization better. Leaving it as is for now.

**Review by TimCsaky — APPROVED** [review]

**Review by kamorel — APPROVED** [review]

**Pull request merged**